### PR TITLE
Spark sum can overflow

### PR DESCRIPTION
**Pull request by jinchengchenghh**

### CODE REVIEW

**Review by rui-mo — APPROVED** [review]
I found the behaviour may depend on a Spark config to throw or not. Maybe we can use this fix for now, and in the long term, we can consider to implement sum under spark sql folder to resolves the differences.

File: velox/functions/prestosql/aggregates/SumAggregate.h [hunk 1]
```diff
@@ -148,7 +148,9 @@ class SumAggregate
   template <typename TData>
   static void updateSingleValue(TData& result, TData value) {
     if constexpr (
-        std::is_same_v<TData, double> || std::is_same_v<TData, float>) {
+        std::is_same_v<TData, double> || std::is_same_v<TData, float> ||
+        std::is_same_v<TData, int8_t> || std::is_same_v<TData, int16_t> ||
+        std::is_same_v<TData, int32_t> || std::is_same_v<TData, int64_t>) {
       result += value;
```
Comment (rui-mo) [inline velox/functions/prestosql/aggregates/SumAggregate.h:154]: What is the type of reuslt? Maybe accumulated type should be used for the result here.

Comment (rui-mo) [reply velox/functions/prestosql/aggregates/SumAggregate.h:154]: Looks like the accumalted types for int8/int16/int32 are all int64. Maybe we only need to handle int64 here.
```
          case TypeKind::TINYINT:
            return std::make_unique<T<int8_t, int64_t, int64_t>>(BIGINT());
          case TypeKind::SMALLINT:
            return std::make_unique<T<int16_t, int64_t, int64_t>>(BIGINT());
          case TypeKind::INTEGER:
            return std::make_unique<T<int32_t, int64_t, int64_t>>(BIGINT());
          case TypeKind::BIGINT:
            return std::make_unique<T<int64_t, int64_t, int64_t>>(BIGINT());
```

Comment (jinchengchenghh) [reply velox/functions/prestosql/aggregates/SumAggregate.h:153]: Fixed

**Comment by jinchengchenghh** [comment]
Yes, you are right, it is controled by ansi enabled

**Comment by jinchengchenghh** [comment]
Mark gere, after switch to spark function, we should check AverageAggregate and select add numeric overflow format

**Pull request merged**